src/Sarif/Visitors/SarifVersionOneToCurrentVisitor.cs

```diff
@@ -243,21 +243,106 @@ public Rule TransformRuleVersionOne(RuleVersionOne node)
             return rule;
         }
 
+        private List<FileLocation> CreateFileLocationsFromDictionary(IDictionary<string, string> dictionary, Run run)
+        {
+            List<FileLocation> result = null;
+
+            if (dictionary != null)
+            {
+                result = new List<FileLocation>();
+
+                foreach (string key in dictionary.Keys)
+                {
+                    var fileLocation = new FileLocation
+                    {
+                        Uri = new Uri(key, UriKind.RelativeOrAbsolute)
+                    };
+                    result.Add(fileLocation);
+
+                    if (run != null && !string.IsNullOrWhiteSpace(dictionary[key]))
+                    {
+                        // We have contents, so mention this file in run.files
+                        if (run.Files == null)
+                        {
+                            run.Files = new Dictionary<string, FileData>();
+                        }
+
+                        if (!run.Files.ContainsKey(key))
+                        {
+                            run.Files.Add(key, new FileData
+                            {
+                                Contents = new FileContent
+                                {
+                                    Text = dictionary[key]
+                                },
+                                FileLocation = fileLocation
+                            });
+                        }
+                    }
+                }
+            }
+
+            return result;
+        }
+
+        public Invocation TransformInvocationVersionOne(InvocationVersionOne node)
+        {
+            Invocation invocation = null;
+
+            if (node != null)
+            {
+                invocation = new Invocation
+                {
+                    Account = node.Account,
+                    CommandLine = node.CommandLine,
+                    EndTime = node.EndTime,
+                    EnvironmentVariables = node.EnvironmentVariables,
+                    Machine = node.Machine,
+                    ProcessId = node.ProcessId,
+                    Properties = node.Properties,
+                    ResponseFiles = CreateFileLocationsFromDictionary(node.ResponseFiles, SarifLog.Runs.Last()),
+                    StartTime = node.StartTime,
+                    WorkingDirectory = node.WorkingDirectory
+                };
+
+                if (!string.IsNullOrWhiteSpace(node.FileName))
```

Comment:
Remove this test. `invocation.commandLine` is allowed to exist even if `invocation.fileName` does not.

Those properties exist for two different reasons. `invocation.fileName`  (which is the improved name `executableLocation` in v2) is meant to allow the consumer to know exactly where the tool was executed from (like, it might help you discover that you'd executed an old version from some directory earlier in `PATH`). It doesn't have to be there, so don't couple these together.

Reply:
Sorry, I misspoke. I meant "don't condition your parsing of `commandLine` on whether this _other_ property `fileName` is present". But that turned out to be moot, because you weren't going to parse the command line itself.

You _do_ have to test `fileName` before trying to make a `Uri` out of it. Sorry again.